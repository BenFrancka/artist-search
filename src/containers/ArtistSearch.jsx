import React, { useState, useEffect } from 'react';
import ArtistList from '../components/ArtistList.jsx';
import SearchControls from '../components/SearchControls';
import { fetchArtists } from '../services/fetchUtils';
import Pagination from 'react-js-pagination';

const ArtistSearch = () => {
  const [artistList, setArtistList] = useState([]);
  const [loading, setLoading] = useState(false);
  const [searchTerm, setSearchTerm] = useState('doors');

  useEffect(() => {
    fetchArtists(searchTerm).then((res) =>
      setArtistList(res)
    );
  }, [searchTerm]);

  const handleSearch = ({ target }) => {
    setSearchTerm(target.value);
  };

  return (
    <>
      <SearchControls searchTerm={searchTerm} onSearchChange={handleSearch} />

      <Pagination
        activePage={activePage}
        itemsCountPerPage={25}
        onChange={handlePageChange}
      />

      {artistList.length && <ArtistList artists={artistList} />}
    </>
  );
};
export default ArtistSearch;
