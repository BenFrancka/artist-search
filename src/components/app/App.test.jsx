import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import ArtistSearch from '../../containers/ArtistSearch';
import { MemoryRouter } from 'react-router';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import test from '../../fixtures/test.json';

const server = setupServer(
  rest.get('http://musicbrainz.org/ws/2/artist', (req, res, ctx) => {
    return res(ctx.json(test));
  })
);

describe('AristSearch Container', () => {
  beforeAll(() => server.listen());
  afterAll(() => server.close());

  it('should display a list of artists matching a search term', async () => {
    const { container } = render(
      <MemoryRouter>
        <ArtistSearch />
      </MemoryRouter>
    );

    const searchBar = await screen.findByRole('textbox', { name: 'searchBar' });
    userEvent.type(searchBar, '{selectall}{del}beatles');

    return waitFor(() => {
      const list = screen.getAllByText('beatles', { exact: false });

      expect(list.length).toBeGreaterThan(2);
      expect(container).toMatchSnapshot();
    });
  });
});
