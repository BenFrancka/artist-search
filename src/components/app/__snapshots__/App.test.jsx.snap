// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AristSearch Container should display a list of artists matching a search term 1`] = `
<div>
  <label
    for="search"
  >
    Search:
    <input
      aria-label="searchBar"
      name="searchTerm"
      placeholder="Find an Artist"
      type="text"
      value="beatles"
    />
  </label>
  <button
    disabled=""
  >
    Prev.
  </button>
  <button
    disabled=""
  >
    Next.
  </button>
  <ul
    aria-label="artists"
  >
    <a
      href="/releases/The Beatles/b10bbbfc-cf9e-42e0-be17-e2c3e1d2600d"
    >
      <li>
        <p>
          The Beatles
        </p>
      </li>
    </a>
    <a
      href="/releases/The Tape-beatles/5e685f9e-83bb-423c-acfa-487e34f15ffd"
    >
      <li>
        <p>
          The Tape-beatles
        </p>
      </li>
    </a>
    <a
      href="/releases/Blues Beatles/1019b551-eba7-4e7c-bc7d-eb427ef54df2"
    >
      <li>
        <p>
          Blues Beatles
        </p>
      </li>
    </a>
    <a
      href="/releases/Instrumental Beatles/5a45e8c5-e8e5-4f05-9429-6dd00f0ab50b"
    >
      <li>
        <p>
          Instrumental Beatles
        </p>
      </li>
    </a>
    <a
      href="/releases/Sex Beatles/e897e5fc-2707-49c8-8605-be82b4664dc5"
    >
      <li>
        <p>
          Sex Beatles
        </p>
      </li>
    </a>
    <a
      href="/releases/Powdered Beatles/74e70126-def2-4b76-a001-ed3b96080e24"
    >
      <li>
        <p>
          Powdered Beatles
        </p>
      </li>
    </a>
    <a
      href="/releases/Tokyo Beatles/bc569a61-dd62-4758-86c6-e99dcb1fdda6"
    >
      <li>
        <p>
          Tokyo Beatles
        </p>
      </li>
    </a>
    <a
      href="/releases/Beatles Chillout/3133aeb8-9982-4e11-a8ff-5477996a80bf"
    >
      <li>
        <p>
          Beatles Chillout
        </p>
      </li>
    </a>
    <a
      href="/releases/Counterfeit Beatles/ad60d963-44f1-4b41-b785-8284edcaaffe"
    >
      <li>
        <p>
          Counterfeit Beatles
        </p>
      </li>
    </a>
    <a
      href="/releases/Shitty Beatles/de0769fa-7c32-4706-9c8c-03631c90f208"
    >
      <li>
        <p>
          Shitty Beatles
        </p>
      </li>
    </a>
    <a
      href="/releases/Beatles Back2Back/7cac6d47-ef4e-4347-8835-63ed3f2e74a7"
    >
      <li>
        <p>
          Beatles Back2Back
        </p>
      </li>
    </a>
    <a
      href="/releases/Fab Beatles/dc146c82-7709-41ee-8322-0d4c9fea0378"
    >
      <li>
        <p>
          Fab Beatles
        </p>
      </li>
    </a>
    <a
      href="/releases/Capital Beatles/91a7ade0-d51d-4f94-81e8-44b75e67b862"
    >
      <li>
        <p>
          Capital Beatles
        </p>
      </li>
    </a>
    <a
      href="/releases/Them Beatles/35574687-3a4d-4b30-a01a-43fea73b3430"
    >
      <li>
        <p>
          Them Beatles
        </p>
      </li>
    </a>
    <a
      href="/releases/Norwegian Beatles/3badd533-4ce6-4fd6-a5ee-76799406c355"
    >
      <li>
        <p>
          Norwegian Beatles
        </p>
      </li>
    </a>
    <a
      href="/releases/Daft Beatles/5d25dbfb-7558-45dc-83dd-6d1176090974"
    >
      <li>
        <p>
          Daft Beatles
        </p>
      </li>
    </a>
    <a
      href="/releases/The Beatles Revival Band/0f697bc6-6df7-41c9-b550-39981e520d70"
    >
      <li>
        <p>
          The Beatles Revival Band
        </p>
      </li>
    </a>
    <a
      href="/releases/The Silver Beetles/b06f5860-ce8f-4d8f-92ee-6c6f165ef8ef"
    >
      <li>
        <p>
          The Silver Beetles
        </p>
      </li>
    </a>
    <a
      href="/releases/Henrique Cazes Quarteto/c271056b-1ca6-49b6-918e-f8f039512f6a"
    >
      <li>
        <p>
          Henrique Cazes Quarteto
        </p>
      </li>
    </a>
    <a
      href="/releases/The Beatles Connection/898e211b-06ad-460b-85ee-cd6ac337ef79"
    >
      <li>
        <p>
          The Beatles Connection
        </p>
      </li>
    </a>
    <a
      href="/releases/Pete Best Beatles/d6e9e3d2-b526-4664-8b00-7e0b28b6ed58"
    >
      <li>
        <p>
          Pete Best Beatles
        </p>
      </li>
    </a>
    <a
      href="/releases/The Better Beatles/7383113a-2b44-4fd1-a2b1-277ca68803b7"
    >
      <li>
        <p>
          The Better Beatles
        </p>
      </li>
    </a>
    <a
      href="/releases/The Wife-Beatles/c658db41-358b-43ba-a06b-8a07ab410407"
    >
      <li>
        <p>
          The Wife-Beatles
        </p>
      </li>
    </a>
    <a
      href="/releases/Sgt. Peppercorn's Beatles Marathon/7d5b2854-ce8f-4d4a-a099-364c1b2488d1"
    >
      <li>
        <p>
          Sgt. Peppercorn's Beatles Marathon
        </p>
      </li>
    </a>
    <a
      href="/releases/The Black Beatles/4c3d0136-9235-4c50-b136-be49d17163df"
    >
      <li>
        <p>
          The Black Beatles
        </p>
      </li>
    </a>
  </ul>
</div>
`;
